hibernate-envers/src/main/java/org/hibernate/envers/configuration/internal/metadata/MetadataTools.java

```diff
@@ -285,38 +297,68 @@ public static Element createJoin(
 		return joinMapping;
 	}
 
-	public static void addColumns(Element anyMapping, Iterator selectables) {
+	public static void addColumns(Element anyMapping, Iterator<?> selectables, Metadata metadata) {
```

Comment:
I checked the codebase and found we can improve our generics usage here by changing the second parameter type from `Iterator` to `Iterator<Selectable>`.

Reply:
That leads to a compilation failure inside `AuditMetadataGenerator` at line 636 where we pass in an explicit `Iterator<Column>` argument rather than `Iterator<Selectable>`; hence the `?`.

Reply:
Thanks for the response